src/journal/journald-console.c

```diff
@@ -72,9 +72,9 @@ void server_forward_console(
         /* First: timestamp */
         if (prefix_timestamp()) {
                 assert_se(clock_gettime(CLOCK_MONOTONIC, &ts) == 0);
-                xsprintf(tbuf, "[%5"PRI_TIME".%06ld] ",
+                xsprintf(tbuf, "[%5"PRI_TIME".%06lu] ",
                          ts.tv_sec,
-                         ts.tv_nsec / 1000);
+                         (unsigned long)ts.tv_nsec / 1000);
```

Comment:
I'm very much opposed to casting a 64 bit variable to 32 bits, even if it only affects one fringe architecture.

OK, that'll work too. `1LLU` is 64 bits on archs, afaict. So the two solutions would be equivalent. Maybe the cast is a bit more explicit.